src/components/molecules/TableHeader/components/EditTableTitleModal/EditTableTitleModal.tsx

```diff
@@ -0,0 +1,170 @@
+}
+
+export const EditTableTitleModal: React.FC<EditTableTitleModalProps> = ({
+  isShown,
+  venueName,
+  title,
+  tables,
+  tableOfUser,
+  subtitle,
+  capacity,
+  onHide,
+}) => {
+  const { user } = useUser();
+  const venueId = useVenueId();
+
+  const formDefaultValues = useMemo(
+    () => ({
+      title,
+      subtitle,
+      capacity,
+    }),
+    [capacity, subtitle, title]
+  );
+
+  const {
+    register,
+    handleSubmit,
+    errors,
+    watch,
+    reset,
+  } = useForm<EditTableForm>({
+    mode: "onBlur",
+    reValidateMode: "onBlur",
+    defaultValues: formDefaultValues,
+  });
+
+  // Updates the default values whenever they change. This is required because they are cached after the first render.
+  useEffect(() => {
+    reset(formDefaultValues);
+  }, [formDefaultValues, reset, title]);
+
+  // use useAsyncFn for easier error handling, instead of state hook
+  const [{ error }, updateTables] = useAsyncFn(
+    async (values: EditTableForm) => {
+      if (!user || !tableOfUser || !venueId) return;
+
+      const updatedTable = {
+        ...tableOfUser,
+        title: values.title,
+        subtitle: values.subtitle,
+        capacity: values.capacity,
+      };
+
+      return await firebase.functions().httpsCallable("venue-updateTables")({
+        venueId,
+        tables,
+        updatedTable,
+      });
+    },
+    [tableOfUser, tables, user, venueId]
+  );
+
+  const values = watch();
+
+  const onSubmit = useCallback(async () => {
+    await updateTables(values).then(() => {
+      onHide();
+    });
+  }, [onHide, updateTables, values]);
+
+  return (
+    <Modal show={isShown} onHide={onHide}>
+      <Modal.Body>
+        <form onSubmit={handleSubmit(onSubmit)} className="EditTableTitleModal">
+          <InputField
+            ref={register({ required: true })}
+            name="title"
+            defaultValue={title}
+            containerClassName="EditTableTitleModal__input--spacing"
+            placeholder="Table topic"
+          />
+          {errors.title && errors.title.type === "required" && (
+            <span className="input-error">Topic is required</span>
+          )}
+
+          <InputField
+            ref={register}
+            name="subtitle"
+            containerClassName="EditTableTitleModal__input--spacing"
+            placeholder="Describe this table (optional)"
+          />
+
+          <div className="EditTableTitleModal__capacity">
+            <label className="EditTableTitleModal__max-capacity">
+              Number of seats (max {MAX_TABLE_CAPACITY})
+            </label>
+            <InputField
+              ref={register({
+                required: true,
+                max: MAX_TABLE_CAPACITY,
+                min: MIN_TABLE_CAPACITY,
+              })}
+              name="capacity"
+              type="number"
+              min={MIN_TABLE_CAPACITY}
+              max={MAX_TABLE_CAPACITY}
+              placeholder="Max seats"
+            />
```

Comment:
`input` fields need to be nested within their associated `label`, or use an `id` to bind them together. We prefer the nesting since `id`'s aren't very React'y.

```suggestion
            <label className="EditTableTitleModal__max-capacity">
              Number of seats (max {MAX_TABLE_CAPACITY})

              <InputField
                ref={register({
                  required: true,
                  max: MAX_TABLE_CAPACITY,
                  min: MIN_TABLE_CAPACITY,
                })}
                name="capacity"
                type="number"
                min={MIN_TABLE_CAPACITY}
                max={MAX_TABLE_CAPACITY}
                placeholder="Max seats"
              />
            </label>
```

**Further Reading:**

- https://developer.mozilla.org/en-US/docs/Web/HTML/Element/label
  - > To associate the `<label>` with an `<input>` element, you need to give the `<input>` an `id` attribute. The `<label>` then needs a `for` attribute whose value is the same as the `input`'s `id`.
  - > **Alternatively, you can nest the `<input>` directly inside the `<label>`, in which case the `for` and `id` attributes are not needed because the association is implicit.**

Reply:
Is there any point to us adding both the `useForm` based validations, as well as the native browser validations? Should we just pick one or the other?

Personally I feel like it will be simpler/cleaner/less error prone for us to pick one; and given we use it everywhere/it's more powerful, I would probably be leaning towards `useForm`'s version rather than the native browser fields.

What do we get from the native that we don't/can't get from `useForm`'s version? And are there ways we can achieve those same objectives by slightly tweaking how we use `useForm`'s validations?

(also see my comments on a previous review thread about this, seemingly `useForm` had an option to apply the native validations in `v3`, but it was dropped in `v4+` since not many people bothered to use it)

---

If you are going to set them natively, you're missing the `required` prop:

- https://developer.mozilla.org/en-US/docs/Learn/Forms/Form_validation#using_built-in_form_validation

(which highlights how easy it can be to make mistakes when we're duplicating the logic across 2 different validation methods)

Reply:
I'm using it here natively, only because the numeric input has those 2 buttons for increment and decrement.

Looks like they can be removed using this: https://www.w3schools.com/howto/howto_css_hide_arrow_number.asp

If I do this, I can remove the native validation and we can stick with the `useForm` and it's consistency.

@0xdevalias How does this sound?

Reply:
Also, any idea how to add `space-between` when the 2 elements are child and parent?

Reply:
After changing it to:
```tsx
<label>
<input/>
</label>
```
The `space-between` is not working anymore and I can't seem to find the proper css styles.

Reply:
Fixed in: https://github.com/sparkletown/sparkle/pull/1377/commits/0fb129c74777ba4ef198bb41d5c67ebdf5ee66c4